hooks/useFetchStakingPoolsData.js

```diff
@@ -0,0 +1,239 @@
+    name: "getRewardForDuration",
+  });
+  const lastTimeRewardApplicableABI = find(MultiRewards, {
+    name: "lastTimeRewardApplicable",
+  });
+  const periodFinishABI = find(MultiRewards, { name: "rewardData" });
+  const accountBalanceOfABI = find(MultiRewards, { name: "balanceOf" });
+  const earnedABI = find(MultiRewards, { name: "earned" });
+
+  let stakingPoolsFunctions = [];
+  STAKING_POOLS.map(async (stakingPool) => {
+    stakingPoolsFunctions[stakingPool.id] = {
+      // Pool size
+      getBalanceOf: connex?.thor
+        .account(stakingPool.rewardsAddress[VECHAIN_NODE])
+        .method(totalSupplyABI),
+
+      // Pool Reward For Duration
+      getRewardForDuration: connex?.thor
+        .account(stakingPool.rewardsAddress[VECHAIN_NODE])
+        .method(getRewardForDurationABI),
+
+      // Last Time Reward Applicable
+      getLastTimeRewardApplicable: connex?.thor
+        .account(stakingPool.rewardsAddress[VECHAIN_NODE])
+        .method(lastTimeRewardApplicableABI),
+
+      // Period Finish
+      getPeriodFinish: connex?.thor
+        .account(stakingPool.rewardsAddress[VECHAIN_NODE])
+        .method(periodFinishABI),
+
+      //  Current stake
+      getAccountBalanceOf: connex?.thor
+        .account(stakingPool.rewardsAddress[VECHAIN_NODE])
+        .method(accountBalanceOfABI),
+
+      // Unstaked staking token balance
+      getUnstakedBalanceOf: connex?.thor
+        .account(stakingPool.stakingTokenAddress[VECHAIN_NODE])
+        .method(balanceOfABI),
+
+      // Unstaked staking token approval amount
+      getUnstakedAllowanceAmount: connex?.thor
+        .account(stakingPool.stakingTokenAddress[VECHAIN_NODE])
+        .method(allowanceABI),
+
+      // Claimable token
+      getEarned: connex?.thor
+        .account(stakingPool.rewardsAddress[VECHAIN_NODE])
+        .method(earnedABI),
+    };
+  });
+
+  const getPoolData = async () => {
+    return await Promise.all(
+      STAKING_POOLS.map(async (stakingPool) => {
+        // Pool size
+        const {
+          decoded: { 0: poolSize },
+        } = await stakingPoolsFunctions[stakingPool.id].getBalanceOf.call();
+
+        const rewardDataABI = find(MultiRewards, { name: "rewardData" });
+        let method =
+          connexStakingPools[stakingPool.id].rewardsContract.method(
+            rewardDataABI
+          );
+        let res = await method.call(stakingPool.rewardTokens[0].address[VECHAIN_NODE]);
+        const rewardRate = ethers.BigNumber.from(res.decoded.rewardRate);
+
+        const totalSupplyABI = find(IERC20, { name: "totalSupply" });
+        method = connex.thor
+          .account(stakingPool.stakingTokenAddress[VECHAIN_NODE])
+          .method(totalSupplyABI);
+        res = await method.call();
+
+        const totalLPTokenSupply = BigNumber.from(res.decoded[0]);
+        const numberOfLPTokensStaked = poolSize;
+        // This is calculated in the amount of vex tokens instead of usd
+        const tokenTvl =
```

Comment:
confirm this is the TVL of the LP pair not staked in the pool?

Reply:
Correct